readinglist/views/oauth.py

```diff
@@ -74,12 +75,12 @@ def fxa_oauth_token(request):
         token = auth_client.trade_code(code)
     except fxa_errors.OutOfProtocolError:
         return HTTPServiceUnavailable()
-    except fxa_errors.InProtocolError:
-        # XXX: use exception details
-        request.response.status_code = 400
-        return
-
-    data = {
-        'token': token,
-    }
-    return data
+    except fxa_errors.InProtocolError as error:
+        logger.exception(error)
+        request.errors.add(
+            location='querystring',
+            name='code',
+            description='Firefox Account code validation failed.')
+        raise json_error(request.errors)
+
+    return HTTPFound(location='%s?token=%s' % (stored_redirect, token))
```

Comment:
Wouldn't it be easier for http clients if we add a header ? 
Parsing the location header can be painful and error-prone no ?